src/main/webapp/app/exercises/programming/shared/code-editor/build-output/code-editor-build-output.component.ts

```diff
@@ -182,15 +183,14 @@ export class CodeEditorBuildOutputComponent implements AfterViewInit, OnInit, On
 
     /**
      * Decides if the build log should be shown.
-     * If All tests were successful or there is test feedback -> don't show build logs.
-     * Else -> show build logs.
+     * Fetch the build logs only if the submission could not be build
      * @param result
      */
     fetchBuildResults(result: Result | null): Observable<BuildLogEntry[] | null> {
-        if ((result && result.successful) || (result && !result.successful && result.feedbacks && result.feedbacks.length)) {
-            return of([]);
-        } else {
+        if ((result?.submission as ProgrammingSubmission).buildFailed) {
```

Comment:
should we handle the (edge) case that `result?.submission` is not available and then try to get the build logs anyway? In such a case it would be fine if the server throws a 403, because it should not happen. And in case the submission is not available (for whatever reason) and the build has failed, the user experience would be better

Reply:
Sounds good to me